packages/rpc-methods/src/restricted/manageAccounts.ts

```diff
@@ -102,14 +106,11 @@ export function manageAccountsImplementation({
 
     assert(params, SnapMessageStruct);
     const keyring = await getSnapKeyring(origin);
-    const result = await keyring.handleKeyringSnapMessage(origin, params);
-
-    const { method } = params;
-    if (['updateAccount', 'createAccount', 'deleteAccount'].includes(method)) {
-      await saveSnapKeyring();
```

Comment:
IIRC the initial implementation had a callback too and we suggested this implementation instead.

I can understand that we don't want the RPC method to know about the underlying implementation, but I also think passing the a hook around isn't the best.

Hmm. What do you think @Mrtenz ?

Reply:
True, I forgot about it when talking to @danroc earlier today and that was the reason to call `saveSnaoKeyring` inside the method and not pass it.

Discussion: https://github.com/MetaMask/snaps/pull/1290#discussion_r1236572775
Commit: [bbc422d](https://github.com/MetaMask/snaps/pull/1290/commits/bbc422da04d6d9a9cf16b927bd99042a69a5e354)

And since we were short on time to release the feature I didn't try to find a better solution

Reply:
I think we may actually need to pass an object with multiple callback functions, not only `saveSnapKeyring`.

We will soon have a use case where the `SnapKeyring` will need to trigger confirmation screens, which is yet another callback that has to be called from the SnapKeyring.

As I said in the PR, the `snap_manageAccounts` should be just a passthrough method to allow a keyring snap to talk to the `SnapKeyring`. So, in line with your comment, I'll completely remove the callbacks from `rpc-methods`.